### PR TITLE
Add additional tests for RegEx

### PR DESCRIPTION
**Pull request by toastedbreadandomelette**
This PR should hopefully address more tests for RegEx. This task is part of https://github.com/godotengine/godot/issues/43440.

More Details:
- First RegEx test PR: https://github.com/godotengine/godot/pull/43781

### CODE REVIEW

File: modules/regex/tests/test_regex.h [hunk 1]
```diff
@@ -131,6 +131,48 @@ TEST_CASE("[RegEx] Empty Pattern") {
 	CHECK(re.is_valid());
 }
 
+TEST_CASE("[RegEx] Complex Grouping") {
+	const String test = "https://docs.godotengine.org/en/latest/contributing/";
+
+	// Ignored protocol in grouping
```
Comment (AThousandShips) [inline modules/regex/tests/test_regex.h:137]:
```suggestion
	// Ignored protocol in grouping.
```
Per the [comment style](https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html#comment-style-guide).

File: modules/regex/tests/test_regex.h [hunk 2]
```diff
@@ -131,6 +131,48 @@ TEST_CASE("[RegEx] Empty Pattern") {
...
+
+	CHECK(expr->get_string(1) == "docs");
+	CHECK(expr->get_string(2) == "godotengine");
+	CHECK(expr->get_string(3) == "org");
+}
+
+TEST_CASE("[RegEx] Number Expression") {
+	const String test = "(2.5e-3 + 35 + 46) / 2.8e0 = 28.9294642857";
+
+	// Not an exact regex for number but good test
```
Comment (AThousandShips) [inline modules/regex/tests/test_regex.h:154]:
```suggestion
	// Not an exact regex for number but a good test.
```

File: modules/regex/tests/test_regex.h [hunk 3]
```diff
@@ -131,6 +131,48 @@ TEST_CASE("[RegEx] Empty Pattern") {
...
+	RegEx re2("([+-]?\\d+)(\\.\\d+([eE][+-]?\\d+)?)?");
+	REQUIRE(re2.is_valid());
+	Array number_match = re2.search_all(test);
+
+	CHECK (number_match.size() == 5);
```
Comment (AThousandShips) [inline modules/regex/tests/test_regex.h:159]:
```suggestion
	CHECK(number_match.size() == 5);
```

**Comment by AThousandShips** [comment]
Also please squash your commits into one, see [here](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) 

Make sure you push your changes with `git push --force`

File: modules/regex/tests/test_regex.h [hunk 4]
```diff
@@ -131,6 +131,48 @@ TEST_CASE("[RegEx] Empty Pattern") {
...
+	CHECK(expr->get_string(1) == "docs");
+	CHECK(expr->get_string(2) == "godotengine");
+	CHECK(expr->get_string(3) == "org");
+}
+
+TEST_CASE("[RegEx] Number Expression") {
+	const String test = "(2.5e-3 + 35 + 46) / 2.8e0 = 28.9294642857";
+
+	// Not an exact regex for number but a good test.
+	RegEx re2("([+-]?\\d+)(\\.\\d+([eE][+-]?\\d+)?)?");
```
Comment (AThousandShips) [inline modules/regex/tests/test_regex.h:155]:
```suggestion
	RegEx re("([+-]?\\d+)(\\.\\d+([eE][+-]?\\d+)?)?");
```
No need to rename it, just confusing

**Comment by AThousandShips** [comment]
You erased your previous fixes, please add them back in and squash all commits into one

**Comment by toastedbreadandomelette** [comment]
These changes should be as per the suggestions. Let me know if there are more changes.

**Comment by inquisitev** [comment]
> You erased your previous fixes, please add them back in and squash all commits into one

why ask developers to manually squash into one commit when github can do it for you on merge of PR by checking a box under settings?

**Comment by AThousandShips** [comment]
Because we don't do those features, they add their own artifacts and can create issues as far as I know

**Comment by inquisitev** [comment]
> Because we don't do those features, they add their own artifacts and can create issues as far as I know

interesting, we use that feature at work and i have not seen any issues, but ill take your word for it

**Comment by Calinou** [comment]
We don't use GitHub's squash and merge feature on the main Godot repository, since it loses the merge commit in the process. We prefer keeping the merge commit to make the Git history more self-documented (even if that makes it longer when not using the `--no-merges` option).

**Review by Calinou — APPROVED** [review]
Tested locally (rebased on top of `master` https://github.com/godotengine/godot/commit/83b916bb00eb8cac4584143a80f5cfad7ce1e908), it works as expected.

I force-pushed a rebased version with the commit message modified to follow conventions used in the repository, so this should be good to merge now.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**